xarray/namedarray/utils.py

```diff
@@ -194,3 +222,66 @@ def __dask_tokenize__(self) -> Hashable:
         from dask.base import normalize_token
 
         return normalize_token((type(self), self._value))  # type: ignore[no-any-return]
+
+
+def consolidate_dask_from_array_kwargs(
```

Comment:
This whole function might not need to live in `NamedArray`, just in xarray main for backwards compatibility. See my other comment about removing dask-specific kwargs from the NamedArray API.